code/game/objects/machinery/portable_reagent_tank.dm

```diff
@@ -0,0 +1,217 @@
+	playsound(src, 'sound/effects/pop.ogg', 100)
+	update_icon()
+
+/obj/machinery/deployable/reagent_tank/disassemble(mob/user)
+	. = ..()
+	var/obj/item/storage/internal_bag = internal_item
+	for(var/mob/watching in internal_bag.content_watchers)
+		internal_bag.close(watching)
+	playsound(src, 'sound/machines/elevator_openclose.ogg', 50)
+
+/obj/item/storage/reagent_tank
+	name = "portable reagent dispenser"
+	desc = "A large vessel for transporting chemicals. Has a cabinet for storing chemical supplies."
+	icon = 'icons/obj/items/chemistry.dmi'
+	icon_state = "dispenser"
+	item_state_worn = TRUE
+	item_state = "reagent_dispenser"
+	flags_equip_slot = ITEM_SLOT_BACK
+	w_class = WEIGHT_CLASS_HUGE
+	max_w_class = WEIGHT_CLASS_SMALL	//Beaker size
+	storage_slots = null
+	max_storage_space = 5
+	can_hold = list(/obj/item/reagent_containers, /obj/item/reagent_scanner)
+	max_integrity = 200
+	var/container_flags = TRANSPARENT
+	var/max_volume = 3000
+	///List of reagents this dispenser will start with
+	var/list/starting_reagents
+
+/obj/item/storage/reagent_tank/Initialize()
+	. = ..()
+	create_reagents(max_volume, container_flags, starting_reagents)
+	AddElement(/datum/element/deployable_item, /obj/machinery/deployable/reagent_tank, 3 SECONDS, 3 SECONDS)
+	//Comes with a scanner by default so players can scan the tanks
+	new /obj/item/reagent_scanner/adv (src)
+
+/obj/item/storage/reagent_tank/examine(mob/user)
+	. = ..()
+	. += "[span_bold("Ctrl Click")] an adjacent tile to deploy."
+
+/obj/item/storage/reagent_tank/update_overlays()
+	. = ..()
+	if(reagents.total_volume)
+		var/image/filling = image(icon, src, "[icon_state]")
+		var/percent = round((reagents.total_volume/max_volume) * 100)
+		switch(percent)
+			if(0 to 19)
+				filling.icon_state = "[icon_state]_1"
+			if(20 to 39)
+				filling.icon_state = "[icon_state]_2"
+			if(40 to 59)
+				filling.icon_state = "[icon_state]_3"
+			if(60 to 79)
+				filling.icon_state = "[icon_state]_4"
+			if(80 to 99)
+				filling.icon_state = "[icon_state]_5"
+			if(99 to INFINITY)
+				filling.icon_state = "[icon_state]_full"
+		filling.color = mix_color_from_reagents(reagents.reagent_list)
+		. += filling
+
+/obj/item/storage/reagent_tank/on_reagent_change()
+	update_icon()
+
+/obj/item/storage/reagent_tank/attack_hand(mob/living/user)
+	if(CHECK_BITFIELD(flags_item, IS_DEPLOYED))
+		return open(user)
+	return ..()
+
+/obj/item/storage/reagent_tank/open(mob/user)
+	if(CHECK_BITFIELD(flags_item, IS_DEPLOYED))
+		return ..()
+
+/obj/item/storage/reagent_tank/attempt_draw_object(mob/living/user)
+	if(!CHECK_BITFIELD(flags_item, IS_DEPLOYED))
+		balloon_alert(user, "[src] must be deployed!")
+		return FALSE
+	return ..()
+
+/obj/item/storage/reagent_tank/do_quick_equip(mob/user)
+	balloon_alert(user, "[src] must be deployed!")
+
+/obj/item/storage/reagent_tank/can_be_inserted(obj/item/W, warning)
+	if(!CHECK_BITFIELD(flags_item, IS_DEPLOYED))
+		balloon_alert(usr, "Deploy [src] to store [W]!")
```

Comment:
Dont use inserts in balloon alerts like this
The player sees it over src they dont need to be told
Ditto for the rest of the balloon alerts
Also you dont need punctuation